index.html

```diff
@@ -296,141 +292,176 @@
-        if (type == "full") {
+    }
+    if (scheduleObj.teacher == "N/A") {
+      scheduleObj.teacher = "";
+    }
+    if (scheduleObj.room == "N/A") {
+      scheduleObj.room = "";
+    }
+    return scheduleObj;
+  }
 
-          // If the user is a normal user and the class is a normal class
-          if (scheduleObj.teacher != "" && schedule.grade) {
-            var sanitized_teacher_name;
-            sanitized_teacher_name = scheduleObj.teacher.toLowerCase();
-            sanitized_teacher_name = sanitized_teacher_name.replace(/ /g,"_");
-            scheduleObj.avatar = "/96x96_photos/" + sanitized_teacher_name + ".jpg";
-            scheduleObj.teacherLink = "/teacher/" + sanitized_teacher_name;
+  // Add the images on the left side to full schedules
+  function addScheduleImages(scheduleObj, userSchedule) {
+    // If the user is a normal user and the class is a normal class
+    if (scheduleObj.teacher != "" && userSchedule.grade) {
+      var sanitized_teacher_name;
+      sanitized_teacher_name = scheduleObj.teacher.toLowerCase();
+      sanitized_teacher_name = sanitized_teacher_name.replace(/ /g,"_");
+      scheduleObj.avatar = "/96x96_photos/" + sanitized_teacher_name + ".jpg";
+      scheduleObj.teacherLink = "/teacher/" + sanitized_teacher_name;
 
-          // If class is a special class or free period
-          } else if (scheduleObj.period == "X" || scheduleObj.name == "Free Period") {
-            scheduleObj.teacherLink = "";
-            var image_url = scheduleObj.name.toLowerCase();
-            image_url = image_url.replace("/", "");
-            image_url = image_url.replace(/\s/g, '');
-            scheduleObj.avatar = "/images/" + image_url + ".jpg";
+    // If class is a special class or free period
+    } else if (scheduleObj.period == "X" || scheduleObj.name == "Free Period") {
+      scheduleObj.teacherLink = "";
+      var image_url = scheduleObj.name.toLowerCase();
+      image_url = image_url.replace("/", "");
+      image_url = image_url.replace(/\s/g, '');
+      scheduleObj.avatar = "/images/" + image_url + ".jpg";
 
-          // If the class is a normal class but the user is a teacher
-          } else {
-            sanitized_teacher_name = schedule.firstname + "_" + schedule.lastname;
-            sanitized_teacher_name = sanitized_teacher_name.toLowerCase();
-            scheduleObj.avatar = "/96x96_photos/" + sanitized_teacher_name + ".jpg";
-          }
-          scheduleObj.roomLink = "/room/" + scheduleObj.room.toLowerCase().replace(/ /g,"_");
+    // If the class is a normal class but the user is a teacher
+    } else {
+      sanitized_teacher_name = userSchedule.firstname + "_" + userSchedule.lastname;
+      sanitized_teacher_name = sanitized_teacher_name.toLowerCase();
+      scheduleObj.avatar = "/96x96_photos/" + sanitized_teacher_name + ".jpg";
+    }
+    scheduleObj.roomLink = "/room/" + scheduleObj.room.toLowerCase().replace(/ /g,"_");
+    return scheduleObj;
```

Comment:
You don't need to return anything here since you are modifying the supplied object.